tasks/initialize.rb

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative '../../ruby_task_helper/files/task_helper.rb'
+require_relative '../lib/cli_helper.rb'
+require 'json'
+require 'open3'
+
+class TerraformInitialize < TaskHelper
+  def init(opts)
+    cli_opts = %w[-no-color]
+    dir = File.expand_path(opts[:dir]) if opts[:dir]
+    cli_opts = cli_opts.join(' ')
+
+    check_init_status = dir ? Dir.exist?("#{dir}/.terraform") : Dir.exist?(File.expand_path('.terraform'))
+
+    if !check_init_status || opts[:reinit]
```

Comment:
What are the consequences of `terraform init` when it has already been initialized? From my limited experience Terraform just seems smart enough to know that it is g2g and just gives a message like `already initialized`.

Reply:
I would be tempted to either make `reinit: true` or to remove that all together.

Reply:
Looks like it goes through the exact same tests either way, CLI output in both situations is very similar. You have to rerun init any time you change your code in ways that require new modules to be downloaded. The reinit option's existence may be of minimal value, since its only really useful during development and while developing the Terraform content you'll be working inside the project directory where it's easy to manually rerun `terraform init`.

Other options may be of more value, e.g. upgrade
 
```
(autope) ody@orihime:[terraform]% terraform init                                                                                                                        
Initializing modules...
- loadbalancer in modules/loadbalancer
- networking in modules/networking

Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "google" (hashicorp/google) 3.4.0...
- Downloading plugin for provider "random" (hashicorp/random) 2.2.1...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.google: version = "~> 3.4"
* provider.random: version = "~> 2.2"

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```
```
(autope) ody@orihime:[terraform]% terraform init                                                                                                                        
Initializing modules...

Initializing the backend...

Initializing provider plugins...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.google: version = "~> 3.4"
* provider.random: version = "~> 2.2"

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

Reply:
I agree, `reinit` is probably not necessary. I think the simple init case solves 99% of problems around this. :+1: